app/Controllers/Course.hs

```diff
@@ -0,0 +1,41 @@
+module Controllers.Course
+    (retrieveCourse, courses, courseInfo, depts) where
+
+import Config (databasePath)
+import Control.Monad.IO.Class (liftIO)
+import qualified Data.Text as T (Text, unlines, unpack)
+import Data.List (sort, nub)
+import Database.Persist (Entity)
+import Database.Persist.Sqlite (SqlPersistM, runSqlite, selectList, entityVal)
+import Database.Tables as Tables (Courses, coursesCode)
+import Happstack.Server.SimpleHTTP (ServerPart, Response, toResponse)
+import Util.Happstack (createJSONResponse)
+import qualified Database.CourseQueries as CourseHelpers (queryCourse, getDeptCourses)
+
+-- | Takes a course code (e.g. \"CSC108H1\") and sends a JSON representation
+-- of the course as a response.
+retrieveCourse :: T.Text -> ServerPart Response
+retrieveCourse = liftIO . CourseHelpers.queryCourse
+
+-- | Builds a list of all course codes in the database.
+courses :: IO Response
```

Comment:
As part of the refactoring, standardize all routes to have (return) type `ServerPart Response` rather than `IO Response`. This will often mean moving `liftIO`s from the routes file into the controller files.